pkg/metrics/metrics.go

```diff
@@ -41,3 +42,15 @@ func Init() (metric.MeterProvider, http.Handler, error) {
 
 	return meterProvider, promhttp.HandlerFor(reg, promhttp.HandlerOpts{}), nil
 }
+
+type ctxKeyType struct{}
+
+var ctxKey ctxKeyType = ctxKeyType{}
+
+func FromContext(ctx context.Context) metric.MeterProvider {
+	return ctx.Value(ctxKey).(metric.MeterProvider)
+}
+
+func With(ctx context.Context, pv metric.MeterProvider) context.Context {
```

Comment:
Maybe call it this:
```suggestion
func WithProvider(ctx context.Context, pv metric.MeterProvider) context.Context {
```
?